pkg/collector/corechecks/servicediscovery/module/envs.go

```diff
@@ -15,39 +15,22 @@ import (
 	"strings"
 
 	"github.com/DataDog/datadog-agent/pkg/util/kernel"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
 	"github.com/shirou/gopsutil/v3/process"
 )
 
 const (
-	// injectorMemFdName is the name the injector (Datadog/auto_inject) uses.
-	injectorMemFdName = "dd_environ"
-	injectorMemFdPath = "/memfd:" + injectorMemFdName + " (deleted)"
+	injectorMemFdName = "dd_process_inject_info.msgpack"
+	injectorMemFdPath = "/memfd:" + injectorMemFdName
```

Comment:
aren't you missing the ` (deleted)` part?

Reply:
I see you changed to testing the prefix, wouldn't it be safer to test for the exact full filename?

Reply:
I removed the `" (deleted)"` as it looks like a quirk, will an anonymous file always be marked as deleted? 
Especially that it wasn't ever deleted